packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -59,6 +53,53 @@ interface TableRowExpansionToggleEventDetail {
 
 Note: This event will _not_ be emitted for group rows.
 
+### `Table` API for specifying lazy loaded hierarchical data
+
+In some cases, a client might know that a given row has children, but for performance reasons, that data hasn't been loaded from the backend and given to the table. To enable this use case, there will be an API on the table to specify that a given row should be expandable even if it has no children within the data and to specify that a row is loading its children. That API will look as follows:
+
+```ts
+interface TableRowOptions {
```

Comment:
Are the states mutually exclusive? (i.e. is it invalid for a row be both forced expandable and loading?) If so, do we think that will likely remain true for future row options we add?

In the event that both of those are true, we could consider a couple possible alternative APIs:

1. provide separate methods for each state a row can be in:
`setRowsForceExpandable(id: string[])`
`setRowsLoading(id: string[])`
`clearRowsOptions(id: string[])` (don't love this name but we can workshop it if you like the direction)

2. change the state to be an enum:
`setRowOptions({id: string, option: 'expandable' | 'loading' | 'normal'}[])

Both of those would allow us not to expose the `TableRowOptions` type and would mitigate ambiguous cases like setting multiple incompatible states.

But option 1 would require multiple calls if you wanted to set disparate row states and it would break down if we want to have options that aren't mutually exclusive.

I'm pretty neutral on these ideas and am ok with the current proposal, especially if we have visions of growing the set of options, but wanted to suggest the ideas in case others can see a strong reason to change direction.

Reply:
To try to answer your questions (without being able to predict the future):
> Are the states mutually exclusive?
Yes, depending on how we implement things, the states can be mutually exclusive.

> If so, do we think that will likely remain true for future row options we add?
No, I don't think future states will be mutually exclusive. For example, we might have an error message we want to set on a row through the options while also having it be force expandable.

Thinking through the questions you've asked and the alternatives you proposed, @atmgrifter00 and I have come up with a different proposal that is now reflected in HLD. We think that will make things less ambiguous and help make things more straightforward for our clients.